### PR TITLE
Run binding tests with extreme_assertions

### PR DESCRIPTION
**Pull request by qinsoon**
This PR adds workflows for running bindings with extreme_assertions.
* Add `may_trace_duplicate_edges` in plan constraints. Some plans may trace duplicate edges, and for those plan, we do not run duplicate edge check.
* Add the missing side metadata sanity check for PageProtect.
* Increase timeout for binding tests (as they now include tests with extreme_assertions).

Related PRs:
* https://github.com/mmtk/mmtk-openjdk/pull/95
* https://github.com/mmtk/mmtk-jikesrvm/pull/79

### CODE REVIEW

**Comment by qinsoon** [comment]
openjdk-binding-test
OPENJDK_BINDING_REF=test-extreme-assertions

**Comment by qinsoon** [comment]
jikesrvm-binding-test
JIKESRVM_BINDING_TEST=test-extreme-assertions

**Review by wenyuzhao — APPROVED** [review]
LGTM

File: src/plan/pageprotect/global.rs [hunk 1]
```diff
@@ -168,6 +168,17 @@ impl<VM: VMBinding> PageProtect<VM> {
                 &CONSTRAINTS,
                 global_metadata_specs,
             ),
+        };
+
+        {
+            use crate::util::metadata::side_metadata::SideMetadataSanity;
+            let mut side_metadata_sanity_checker = SideMetadataSanity::new();
+            ret.common
+                .verify_side_metadata_sanity(&mut side_metadata_sanity_checker);
```
Comment (caizixian) [inline src/plan/pageprotect/global.rs:177]: Is this part of the extreme assertions or is it a cheap check?

Comment (qinsoon) [reply src/plan/pageprotect/global.rs:177]: It is only called once per plan at boot time. I guess it is fine. `extreme_assertions` needs this check, but this check itself is not part of `extreme_assertions`.

Comment (caizixian) [reply src/plan/pageprotect/global.rs:177]: Maybe we should add some comments so that when people write new plans, they know to call `SideMetadataSanity`

Comment (qinsoon) [reply src/plan/pageprotect/global.rs:177]: Yeah. I will do that.

Comment (qinsoon) [reply src/plan/pageprotect/global.rs:177]: Sounds good. I have added some comments. But as `new()` is specific for each plan, I can only put some comments on the `Plan` trait type about a few things developers need to do in a plan constructor. I could not think of a better place to put those comments. If you think there is a better place to put those comments, please let me know.

**Review by caizixian — APPROVED** [review]

File: src/plan/pageprotect/global.rs (hunk 1 above)
Comment (k-sareen) [reply src/plan/pageprotect/global.rs:177]: Potentially also the tutorial?

Comment (qinsoon) [reply src/plan/pageprotect/global.rs:179]: The tutorial actually includes the constructor code that shows the side metadata sanity checker. I have added more comments to all the plan constructors (including the tutorial plan).

**Pull request merged**